extensions/amp-lightbox-viewer/0.1/service/lightbox-manager-impl.js

```diff
@@ -120,20 +135,57 @@ export class LightboxManager {
    * @private
    */
   processLightboxElement_(element) {
-    if (element.tagName == 'AMP-CAROUSEL') {
-      // TODO: special case carousel
+    if (element.tagName.toLowerCase() == CAROUSEL_TAG) {
+      let lightboxGroupId = element.getAttribute('lightbox');
+      if (!lightboxGroupId) {
+        lightboxGroupId = 'carousel'
+          + (element.getAttribute('id') || this.counter_++);
+      }
+      this.getSlidesFromCarousel_(element).forEach(slide => {
+        // TODO: review naming conventions for component attributes
+        if (!slide.hasAttribute('lightbox-exclude')) {
+          slide.setAttribute('lightbox', lightboxGroupId);
+          this.processBaseLightboxElement_(slide, lightboxGroupId);
+        }
+      });
     } else {
       const lightboxGroupId = element.getAttribute('lightbox') || 'default';
-      if (!this.lightboxGroups_[lightboxGroupId]) {
-        this.lightboxGroups_[lightboxGroupId] = [];
-      }
-      this.lightboxGroups_[lightboxGroupId]
-          .push(dev().assertElement(element));
-      if (this.meetsHeuristicsForTap_(element)) {
-        const viewer = elementByTag(this.ampdoc_.getRootNode(), VIEWER_TAG).id;
-        element.setAttribute('on', 'tap:' + viewer.id + '.activate');
-      }
+      this.processBaseLightboxElement_(element, lightboxGroupId);
+    }
+  }
+
+  processBaseLightboxElement_(element, lightboxGroupId) {
+    if (!this.lightboxGroups_[lightboxGroupId]) {
+      this.lightboxGroups_[lightboxGroupId] = [];
+    }
+    this.lightboxGroups_[lightboxGroupId]
+        .push(dev().assertElement(element));
+    if (this.meetsHeuristicsForTap_(element)) {
+      const viewer = elementByTag(this.ampdoc_.getRootNode(), VIEWER_TAG);
+      element.setAttribute('on', 'tap:' + viewer.id + '.activate');
+    }
+  }
+
+  /**
+   * @param {!Element} element
+   * @return {!Array<!Element>}
+   * @private
+   */
+  getSlidesFromCarousel_(element) {
+    const type = element.getAttribute('type');
+    let carouselSlides = [];
+    if (type == 'slides') {
+      iterateCursor(
+          scopedQuerySelectorAll(element, SLIDE_ITEM_SELECTOR),
+          slide => carouselSlides.push(childElement(slide, () => true))
+      );
+    } else if (type == 'carousel') {
+      const container = dev().assertElement(
+          scopedQuerySelector(element, CAROUSEL_CONTAINER_SELECTOR)
+      );
+      carouselSlides = childElements(container, () => true);
```

Comment:
You could use the [`Element.children`](https://developer.mozilla.org/en-US/docs/Web/API/ParentNode/children) property instead.